### ISSUE TITLE
Confusions about the mnist siamese network example

### ISSUE THREAD

**Issue by qianlongzju** [keras-team/keras#13018]
I have added a callback which uses compute_accuracy to print the accuracy. The result is like the following:

Epoch 31/50
4108/4108 [==============================] - 29s 7ms/step - loss: 0.2519 - accuracy: 0.5166 - val_loss: 0.2533 - val_accuracy: 0.5000
* Accuracy on training set: 77.19%
* Accuracy on test set: 75.29%

The point is that, the result of the function **compute_accuracy** is not matched with **accuracy**.  The logic of these two functions is the same, but implemented in two ways, numpy and keras.  So why the result is not matched?

I create a gist of my code:
https://gist.github.com/qianlongzju/0fa904c7e6b724ff624dab4a8c1d5ae3

Another question is 
I see the code labels the positive pair as **1**, and negative pair as **0**, but why in the **accuracy** function, values less then 0.5 is considered as **True**?
